langchain4j-mistral-ai/src/test/java/dev/langchain4j/model/mistralai/MistralAiChatModelIT.java

```diff
@@ -205,5 +235,224 @@ void should_generate_answer_using_model_medium_and_return_token_usage_and_finish
+                .name("retrieve-payment-date")
+                .description("Retrieve Payment Date")
+                .addParameter("transactionId", STRING)
+                .build();
+
+        List<ChatMessage> chatMessages = new ArrayList<>();
+        UserMessage userMessage = userMessage("What is the payment date of transaction T123?");
+        chatMessages.add(userMessage);
+
+        // when
+        Response<AiMessage> response = mistralLargeModel.generate(singletonList(userMessage), retrievePaymentDate);
+
+        // then
+        AiMessage aiMessage = response.content();
+        assertThat(aiMessage.text()).isNull();
+        assertThat(aiMessage.toolExecutionRequests()).hasSize(1);
+
+        ToolExecutionRequest toolExecutionRequest = aiMessage.toolExecutionRequests().get(0);
+        assertThat(toolExecutionRequest.name()).isEqualTo("retrieve-payment-date");
+        assertThat(toolExecutionRequest.arguments()).isEqualToIgnoringWhitespace("{\"transactionId\":\"T123\"}");
+
+        TokenUsage tokenUsage = response.tokenUsage();
+        assertThat(tokenUsage.inputTokenCount()).isEqualTo(79);
+        assertThat(tokenUsage.outputTokenCount()).isGreaterThan(0);
+        assertThat(tokenUsage.totalTokenCount())
+                .isEqualTo(tokenUsage.inputTokenCount() + tokenUsage.outputTokenCount());
+
+        assertThat(response.finishReason()).isEqualTo(TOOL_EXECUTION);
+        chatMessages.add(aiMessage);
+
+        // given
+        ToolExecutionResultMessage toolExecutionResultMessage = ToolExecutionResultMessage.from(toolExecutionRequest, "{\"date\": \"2024-03-11\"}");
+        chatMessages.add(toolExecutionResultMessage);
+
+        // when
+        Response<AiMessage> response2 = mistralLargeModel.generate(chatMessages);
+
+        // then
+        AiMessage aiMessage2 = response2.content();
+        assertThat(aiMessage2.text()).containsIgnoringCase("T123");
+        assertThat(aiMessage2.text()).containsIgnoringWhitespaces("March 11, 2024");
+        assertThat(aiMessage2.toolExecutionRequests()).isNull();
+
+        TokenUsage tokenUsage2 = response2.tokenUsage();
+        assertThat(tokenUsage2.inputTokenCount()).isEqualTo(78);
+        assertThat(tokenUsage2.outputTokenCount()).isGreaterThan(0);
+        assertThat(tokenUsage2.totalTokenCount())
+                .isEqualTo(tokenUsage2.inputTokenCount() + tokenUsage2.outputTokenCount());
+
+        assertThat(response2.finishReason()).isEqualTo(STOP);
+    }
+
+    @Test
+    void should_return_valid_json_object(){
+
+        // given
+        String userMessage = "Return JSON with two fields: transactionId and status with the values T123 and paid.";
+
+        String expectedJson = "{\"transactionId\":\"T123\",\"status\":\"paid\"}";
+
+        ChatLanguageModel mistralLargeModel = MistralAiChatModel.builder()
+                .apiKey(System.getenv("MISTRAL_AI_API_KEY"))
+                .modelName(MistralAiChatModelName.MISTRAL_LARGE_LATEST.toString())
```

Comment:
minor: would be great to also accept enuns in addition to strings for `modelName`

Reply:
Good idea, I hadn't thought about it, I'll see how to add it

Reply:
Thanks, you can check `OpenAiChatModel` as an example

Reply:
great! I added `responseFormat` param as well to accept enums in addition to strings